sound/soc/intel/boards/sof_es8336.c

```diff
@@ -541,29 +597,41 @@ static int sof_es8336_probe(struct platform_device *pdev)
 	card = &sof_es8336_card;
 	card->dev = dev;
 
+	if (pdev->id_entry && pdev->id_entry->driver_data)
+		quirk = (unsigned long)pdev->id_entry->driver_data;
+
 	/* check GPIO DMI quirks */
 	dmi_check_system(sof_es8336_quirk_table);
 
-	if (!mach->mach_params.i2s_link_mask) {
-		dev_warn(dev, "No I2S link information provided, using SSP0. This may need to be modified with the quirk module parameter\n");
-	} else {
-		/*
-		 * Set configuration based on platform NHLT.
-		 * In this machine driver, we can only support one SSP for the
-		 * ES8336 link, the else-if below are intentional.
-		 * In some cases multiple SSPs can be reported by NHLT, starting MSB-first
-		 * seems to pick the right connection.
-		 */
-		unsigned long ssp = 0;
-
-		if (mach->mach_params.i2s_link_mask & BIT(2))
-			ssp = SOF_ES8336_SSP_CODEC(2);
-		else if (mach->mach_params.i2s_link_mask & BIT(1))
-			ssp = SOF_ES8336_SSP_CODEC(1);
-		else  if (mach->mach_params.i2s_link_mask & BIT(0))
-			ssp = SOF_ES8336_SSP_CODEC(0);
-
-		quirk |= ssp;
+	/* set number of dai links */
+	sof_es8336_card.num_links = 1;
```

Comment:
Isn't this already defined in `static struct snd_soc_card sof_es8336_card {}`?

Reply:
Yes, It is initialized here static struct snd_soc_card sof_es8336_card {.num_links=1,}

Reply:
Ah, nice catch @bardliao 

@reddymur this needs to be handled with a dedicated patch. We need to remove the initialization to one, and always reset the value.
See e.g. how things are done in other machine drivers

````
sof_nau8825.c\0571:	sof_audio_card_nau8825.num_links = 1 + dmic_be_num + hdmi_num;
sof_cs42l42.c\0654:	sof_audio_card_cs42l42.num_links = 1 + dmic_be_num + hdmi_num;
sof_rt5682.c\0912:	sof_audio_card_rt5682.num_links = 1 + dmic_be_num + hdmi_num;
````